ci/kokoro/docker/build-in-docker-cmake.sh

```diff
@@ -126,8 +126,14 @@ if [[ "${CLANG_TIDY:-}" == "yes" && (\
   echo
   io::log_yellow "Running clang-tidy on presubmit build, only changed files are tested."
   ${CMAKE_COMMAND} --build "${BINARY_DIR}" --target nlohmann_json_project
+  io::log "Running clang-tidy for: "
+  # TODO(#3958) - use a simple regular expression like '\.(h|cc)$' when all
+  # targets are clang-tidy clean.
   git diff --name-only "${KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH:-${BRANCH}}" |
-    grep -E '\.(cc|h)$' |
+    grep -E '(\.cc|/(bigquery|firestore|pubsub|spanner)/.*\.h)$' |
+    xargs -r echo
+  git diff --name-only "${KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH:-${BRANCH}}" |
+    grep -E '(\.cc|/(bigquery|firestore|pubsub|spanner)/.*\.h)$' |
     xargs -d '\n' -r -n 1 -P "${NCPU}" clang-tidy -p="${BINARY_DIR}"
```

Comment:
I assumed that the invocation of `clang-tidy` would have it read the `.clang-tidy` file at the root of our repo and ignore all files and headers that it's supposed to ignore. That's not happening? Boo clang-tidy.

You might be able to use the regex at `cat "${PROJECT_ROOT}/.clang-tidy" | grep HeaderFilterRegex`

Reply:
Actually, consider this optional.

Reply:
> I assumed that the invocation of `clang-tidy` would have it read the `.clang-tidy` file at the root of our repo and ignore all files and headers that it's supposed to ignore. That's not happening?

AFAICT, no, it does not happen.

> Boo clang-tidy.

Well, this was always going to be messy.


> You might be able to use the regex at `cat "${PROJECT_ROOT}/.clang-tidy" | grep HeaderFilterRegex`

Done, but please check my math here, regexps in variables scare me.